search/includes/classes/class-versioningcleanupjob.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Automattic\VIP\Search;
+
+class VersioningCleanupJob {
+
+	/**
+	 * The name of the scheduled cron event to run the versioning cleanup.
+	 */
+	const CRON_EVENT_NAME = 'vip_search_versioning_cleanup';
+
+	const WEEK_IN_SECONDS = 7 * 24 * 60 * 60;
+
+	const SEARCH_ALERT_SLACK_CHAT = '#vip-go-es-alerts';
+
+	public function __construct( $indexables, $versioning ) {
+		$this->indexables = $indexables;
+		$this->versioning = $versioning;
+	}
+
+	/**
+	 * Initialize the job class.
+	 *
+	 * @access  public
+	 */
+	public function init() {
+		add_action( self::CRON_EVENT_NAME, [ $this, 'versioning_cleanup' ] );
+
+		$this->schedule_job();
+	}
+
+	/**
+	 * Schedule class versioning job.
+	 *
+	 * Add the event name to WP cron schedule and then add the action.
+	 */
+	public function schedule_job() {
+		if ( ! wp_next_scheduled( self::CRON_EVENT_NAME ) ) {
+			wp_schedule_event( time(), 'weekly', self::CRON_EVENT_NAME );
+		}
+	}
+
+	/**
+	 * Process versioning cleanup.
+	 */
+	public function versioning_cleanup() {
+		$indexables = $this->indexables->get_all();
+
+		foreach ( $indexables as $indexable ) {
+			$inactive_versions = $this->get_inactive_versions( $indexable );
+
+			foreach ( $inactive_versions as $version ) {
+				if ( ! ( $version['active'] ?? false ) ) {
+					// double check that it is not active
+					$this->send_notification( $indexable->slug, $version['number'] );
+				}
+			}
+		}
+	}
+
+
+	/**
+	 * Retrieve inactive versions to be deleted
+	 *
+	 * @param \ElasticPress\Indexable $indexable The Indexable for which to retrieve index versions
+	 * @return array Array of inactive index versions
+	 */
+	public function get_inactive_versions( \ElasticPress\Indexable $indexable ) {
+		$versions = $this->versioning->get_versions( $indexable );
+
+		if ( ! $versions && ! is_array( $versions ) ) {
+			return [];
+		}
+
+		$active_version = null;
+		foreach ( $versions as $version ) {
+			if ( ( $version['active'] ?? false ) && ( $version['activated_time'] ?? false ) ) {
+				$active_version = $version;
+				break;
+			}
+		}
+
+		if ( ! $active_version ) {
+			return [];
+		}
+
+		$week_ago_breakpoint            = time() - self::WEEK_IN_SECONDS;
```

Comment:
We should add a comment to this to remind ourselves why we have this check on the activation time of the active version. It's because we want to keep the old version for a period of time, even if it's older than 1 week, while the new version proves stable.

Reply:
done